nixos/modules/services/audio/spotifyd.nix

```diff
@@ -4,7 +4,11 @@ with lib;
 
 let
   cfg = config.services.spotifyd;
-  spotifydConf = pkgs.writeText "spotifyd.conf" cfg.config;
+  toml = pkgs.formats.toml {};
+  spotifydConf =
+    if cfg.settings != {}
+    then toml.generate "spotify.conf" cfg.settings
+    else lib.trace "Using the stringly typed .config attribute is discouraged. Use the TOML typed .settings attribute instead." (pkgs.writeText "spotifyd.conf" cfg.config);
```

Comment:
The warning is now shown when only `services.spotifyd.enable = true;` is set (I do not think it was before, or perhaps I missed it).